src/Components/Register.js

```diff
@@ -89,13 +79,47 @@ export default function Register() {
     if (loading) return;
     //Registering users without any likes causes /home rendering to error out --> this prevents that.
     if (localUser["likes"]?.length === 0) navigate("/");
-    if (user && forwardToken) {
-      SaveToFirestore(user, localUser).then(() => {
-        navigate("/home");
-      });
-    }
   }, [user, loading, forwardToken]);
 
+  const handleRegister = async (provider) => {
+    try {
+      setRegisterError(null);
+      if (provider === "google") {
+        !user
+          ? await authActions.registerWithGoogle()
+          : await authActions.linkWithGoogle();
```

Comment:
With this logic it's possible to overwrite an existing account - I don't like this as our /register and /login screens look identical, so it seems like an easy mistake to make that could wipe out a bunch of work.

See my suggestion below!